src/robot/Client.ts

```diff
@@ -325,14 +325,26 @@ export class RobotClient implements Robot {
         this.transportFactory = webRTCConn.transportFactory;
 
         webRTCConn.peerConnection.ontrack = (event) => {
-          events.emit('track', event);
-
           const { kind } = event.track;
 
           const eventStream = event.streams[0];
           if (!eventStream) {
+            events.emit('track', event);
             throw new Error('expected event stream to exist');
           }
+
+          // track id has +s to conform to RFC 4566 (https://www.rfc-editor.org/rfc/rfc4566)
+          // where names should not contain colons.
+          const resName = eventStream.id.replaceAll('+', ':');
+          // overriding the stream id to match the resource name and then immediately setting it back to readonly
+          Object.defineProperty(eventStream, 'id', {
+            value: resName,
+            writable: true,
+          });
+          Object.defineProperty(eventStream, 'id', {
+            writable: false,
+          });
```

Comment:
nice find! just curious, can we change the property and make it read-only in one step? or does `read-only` prevent this kind of direct hacking?

```suggestion
            writable: false,
          });
```

Reply:
huh I was having trouble getting to work before but apparently it doesn't even need the `writable: false`